lib/r10k/action/deploy/environment.rb

```diff
@@ -43,6 +45,41 @@ def call
 
         private
 
+        def extract_credentials!
+          if @sshkey_path && @token_path
+            raise R10K::Error, "Cannot specify both an SSH key and a token to use with this deploy."
+          end
+
+          if @sshkey_path
+            if @sshkey_path == '-'
+              sshkey = $stdin.read
+              keypath = write_key_to_tmpfile(sshkey).path
+              { sshkey_file: keypath }
+            elsif File.exists?(@sshkey_path)
+              { sshkey_file: @sshkey_path }
+            else
+              raise R10K::Error, _("{%path} does not exist, cannot load SSH key") % { path: @sshkey_path }
+            end
+          elsif @token_path
+            if @token_path == '-'
+              token = $stdin.read
+            elsif File.exists?(@token_path)
+              token = File.read(@token_path)
+            else
+              raise R10K::Error, _("{%path} does not exist, cannot load OAuth token") % { path: @token_path }
+            end
+            { token: token }
```

Comment:
I assume the choice to send the token itself here, whereas we send the file path for the sshkey, is related to how we'll ultimately use them with rugged?

Reply:
Yes. Rugged expects the token as a string, and a file path for the SSH key.